lib/ansible/modules/network/nxos/nxos_file_copy.py

```diff
@@ -218,28 +356,40 @@ def main():
     file_system = module.params['file_system']
 
     results['transfer_status'] = 'No Transfer'
-    results['local_file'] = local_file
     results['file_system'] = file_system
 
-    if not local_file_exists(module):
-        module.fail_json(msg="Local file {0} not found".format(local_file))
+    if file_pull:
+        src = remote_file.split('/')[-1]
+        local = local_file or src
 
-    dest = remote_file or os.path.basename(local_file)
-    remote_exists = remote_file_exists(module, dest, file_system=file_system)
+        if not module.check_mode:
+            copy_file_from_remote(module, local, file_system=file_system)
```

Comment:
Why is `local_file` necessary here? Are `local_file` and `remote_file` not mutually exclusive?

Reply:
No, remote_file and local_file are not mutually exclusive. Basically when we are copying files, if we want the destination file name to be same as source file name then we can provide just the remote_file. If we want the destination file name to be different, we can give local_file. So this is optional but not mutually exclusive.
The original code (for copying files using scp server on the switch) also has the same logic.

Reply:
Will this line throw an error when `file_pull` is `True` but `local_file` is not set in the playbook?

Reply:
No, because we are passing ```local``` which can be the name of ```local_file``` but if it is not in the playbook, it uses ```remote_file```